src/plugins/intel_gpu/src/graph/impls/ocl/scaled_dot_product_attention.cpp

```diff
@@ -451,9 +451,14 @@ struct scaled_dot_product_attention_impl : multi_stage_primitive<scaled_dot_prod
     static std::unique_ptr<primitive_impl> create(const typed_program_node<scaled_dot_product_attention>& arg, const kernel_impl_params& impl_param) {
         std::vector<kernel_selector::kernel_data> kernels_data;
         auto& kernel_selector = kernel_selector_t::Instance();
-        auto params = impl_param.output_layouts[0].get_partial_shape().size() == 4 ? impl_param : static_canonicalize_shapes(impl_param);
+        bool is_output_rank_4d = impl_param.output_layouts[0].get_partial_shape().size() == 4;
+        auto params = is_output_rank_4d ? impl_param : static_canonicalize_shapes(impl_param);
 
         auto sdpa_kernel_params = get_kernel_params(params, params.is_dynamic());
+        // Known limitation: In vision encoding model of qwen-vl, when the shape of sdpa is 3D and num_heads is 1,
+        // there is an accuracy issue with sdpa_micro kernel. Therefore, it is currently restricted to execute with sdpa_opt kernel.
+        if (!is_output_rank_4d)
```

Comment:
If it is due to num_heads == 1, then the condition should be modified so, what is the actual matter? (rank or head_num)?

Reply:
Anyway  when rank < 4, num_heads will be 1, so I will approve for now, but would like to ask you to check next time, 1) what is the exact root cause of the sdpa_micro failure (really num_heads?) 
2) Or more preferrably, fix the issue in the sdpa_micro kernel itself.

Reply:
Agree with Taylor
@andrew-k-park just to clarify: does the affected case have specific mask configuration or something else, or are we seeing this issue across all 3D SDPAs using sdpa_micro kernel?

Reply:
@sshlyapn It is not confirmed yet that the issue exists across all 3D SDPAs using sdpa_micro kernel. However, 3D SDPA with attention_mask including "-inf" exists in vision encoding model of some qwen-vl models, and there is an issue where the result includes "-nan" when running with sdpa_micro kernel such as LNL/BMG, although there is no issue when running with sdpa_opt kernel like DG2. Therefore, until the issue in sdpa_micro kernel is clearly identified, the use sdpa_opt kernel is restricted in such case. This will be put on the backlog and investigated ASAP.